java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java

```diff
@@ -1144,6 +1147,53 @@ void testCast() {
     }
   }
 
+  @Test
+  void testStringCast() {
+    Integer[] integerArray = {1, 2, 3, 7, 8};
+
+    Short[] shortValues = Arrays.stream(integerArray).map(i -> i.shortValue()).collect(Collectors.toList()).toArray(new Short[integerArray.length]);
+    String[] stringShortValues = Arrays.stream(shortValues).map(i -> String.valueOf(i)).collect(Collectors.toList()).toArray(new String[shortValues.length]);
+
+    testCastFixedWidthToStringsAndBack(DType.INT16, () -> ColumnVector.fromBoxedShorts(shortValues), () -> ColumnVector.fromStrings(stringShortValues));
+
+    String[] stringIntValues = Arrays.asList(integerArray).stream().map(i -> String.valueOf(i)).collect(Collectors.toList()).toArray(new String[integerArray.length]);
+
+    testCastFixedWidthToStringsAndBack(DType.INT32, () -> ColumnVector.fromBoxedInts(integerArray), () -> ColumnVector.fromStrings(stringIntValues));
+
+    Long[] longValues = Arrays.stream(integerArray).map(i -> i.longValue()).collect(Collectors.toList()).toArray(new Long[integerArray.length]);
+    String[] stringLongValues = Arrays.stream(longValues).map(i -> String.valueOf(i)).collect(Collectors.toList()).toArray(new String[longValues.length]);
+
+    testCastFixedWidthToStringsAndBack(DType.INT64, () -> ColumnVector.fromBoxedLongs(longValues), () -> ColumnVector.fromStrings(stringLongValues));
+
+    Float[] floatValues = Arrays.stream(integerArray).map(i -> i.floatValue()).collect(Collectors.toList()).toArray(new Float[integerArray.length]);
+    String[] stringFloatValues = Arrays.stream(floatValues).map(i -> String.valueOf(i)).collect(Collectors.toList()).toArray(new String[floatValues.length]);
+
+    testCastFixedWidthToStringsAndBack(DType.FLOAT32, () -> ColumnVector.fromBoxedFloats(floatValues), () -> ColumnVector.fromStrings(stringFloatValues));
+
+    Double[] doubleValues = Arrays.stream(integerArray).map(i -> i.doubleValue()).collect(Collectors.toList()).toArray(new Double[integerArray.length]);
+    String[] stringDoubleValues = Arrays.stream(doubleValues).map(i -> String.valueOf(i)).collect(Collectors.toList()).toArray(new String[doubleValues.length]);
+
+    testCastFixedWidthToStringsAndBack(DType.FLOAT64, () -> ColumnVector.fromBoxedDoubles(doubleValues), () -> ColumnVector.fromStrings(stringDoubleValues));
+
+    Boolean[] booleans = {true, false, false};
+    String[] stringBools = Arrays.stream(booleans).map(i -> String.valueOf(i)).collect(Collectors.toList()).toArray(new String[booleans.length]);
```

Comment:
This feels like a lot of code just so you don't have to write the same thing but with quotes.
```
Boolean[] booleans = {true, false, false};
String[] stringBools = {"true", "false", "false"};
```

Also could we use generics to make this work?
```
public static <T> String[] convertToString(T[] input) {
  String[] ret = new String[input.length];
  for (int i = 0; i < input.length; i++) {
    ret[i] = String.valueOf(input[i]);
  }
  return ret;
}
```

Reply:
The reason I converted the booleans to stringBools using code was so they both have the same data. Same reason why I chose to convert the int array to other types instead of creating new arrays with values typed in. I might have over thought this one